File: easy/2138. Divide a String Into Groups of Size k/2138. Divide a String Into Groups of Size k.ts
function divideString(s: string, k: number, fill: string): string[] {
    const res: string[] = []
    for (let i = 0; i < s.length; i += k) {
        res.push(s.slice(i, i + k))
    }

    const lastSize = res[res.length - 1].length
    if (lastSize < k) {
        res[res.length - 1] += fill.repeat(k - lastSize)
    }

    return res
};

File: easy/2138. Divide a String Into Groups of Size k/2138. Divide a String Into Groups of Size k.py
from typing import List


class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        res = []
        for i in range(0, len(s), k):
            res.append(s[i : i + k])
        res[-1] += fill * (k - len(res[-1]))
        return res
